server/taskmanager/views.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import logging
+import re
+
+from django.conf import settings
+from django.shortcuts import render, redirect, get_object_or_404
+from rest_framework import mixins, status, viewsets
+from rest_framework.authentication import TokenAuthentication, SessionAuthentication
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from server.auth import CheckAppPermission
+from server.views import JsonQueryFilterBackend, SimpleQueryFilterBackend, deny_restricted_users
+from .models import Pool, Task
+from .serializers import PoolSerializer, TaskSerializer
+
+LOG = logging.getLogger("fm.taskmanager.views")
+
+
+@deny_restricted_users
+def index(request):
+    return redirect('taskmanager:pool-list-ui')
+
+
+@deny_restricted_users
+def list_pools(request):
+    return render(request, 'pool/index.html', {})
+
+
+@deny_restricted_users
+def view_pool(request, pk):
+    pool = get_object_or_404(Pool, pk=pk)
+    return render(request, 'pool/view.html', {"pool": pool, "tc_root_url": settings.TC_ROOT_URL})
+
+
+class PoolViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    API endpoint that allows viewing Pools
+    """
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    permission_classes = (CheckAppPermission,)
+    queryset = Pool.objects.all()
+    serializer_class = PoolSerializer
+    paginate_by_param = 'limit'
+    filter_backends = [
+        JsonQueryFilterBackend,
+        SimpleQueryFilterBackend,
+    ]
+
+
+class TaskViewSet(mixins.ListModelMixin,
+                  mixins.RetrieveModelMixin,
+                  viewsets.GenericViewSet):
+    """
+    API endpoint that allows viewing Tasks
+    """
+    authentication_classes = (TokenAuthentication, SessionAuthentication)
+    permission_classes = (CheckAppPermission,)
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    paginate_by_param = 'limit'
+    filter_backends = [
+        JsonQueryFilterBackend,
+        SimpleQueryFilterBackend,
+    ]
+
+    @action(detail=False, methods=['post'])
+    def update_status(self, request):
```

Comment:
Is this the entry point for machines updating their status? If so, does it require `SessionAuthentication` to be valid? I believe by having that, we allow CSRF to this API.

Reply:
Good catch! Hopefully this isn't too difficult to restrict within the same `ViewSet`. Maybe this is why we had separate views for each action in EC2SpotManager.